doc/mgmt/SONiC_TIMEZONE_HLD.md

```diff
@@ -0,0 +1,546 @@
+      
+**System Clock**    
+    
+This feature also provides the capability to configure the system clock time and date.    
+Note that if the system is synchronized with a NTP server, the manually enterred time will be overriden eventually.    
+The new time and date should be consistent on the host and containers.    
+ 
+**optional items**   
+     
+1. DHCP timezone offset and name options   
+SONiC already request DHCP timezone offset in the existing code. Though backend is not handling it.    
+In addtion, DHCP server can specify the timezone name that the systems should be using as specified in [RFC4833]. E-SONiC does not request this option as of now from the DHCP servers.    
+Either option may force the clients to use the same timezone.    
+
+           
+## 1.1 Requirements
+
+### 1.1.1 Front end configuration and get capabilities
+
+#### 1.1.1.1 add/delete timezone 
+```
+clock timezone US/Hawaii
+```
+Add/delete timezone in the Redis CONFIG DB. Only one timezone can be configured.    
+
+#### 1.1.1.2 Get timezone 
+```
+show clock timezone
+```
+Get the local timezone information from timedatectl output. For example,      
+
+    ```
+    sonic# show clock timezone
+      US/Pacific (PDT, -0700)
+    ```    
+
+    ```
+    root@sonic:~# timedatectl
+          Local time: Thu 2021-07-22 21:32:59 PDT
+      Universal time: Fri 2021-07-23 04:32:59 UTC
+            RTC time: Fri 2021-07-23 04:32:59
+           Time zone: US/Pacific (PDT, -0700)
+    ```
+    
+#### 1.1.1.3 set time and date
+```
+clock set 15:10:06 2021-06-06
+```
+Set time and date.    
+
+#### 1.1.1.4 Get time and date 
+```
+show clock 
+```
+This should display the current local time based on the timezone, time and date user configured. It will be adjusted in case a valid NTP server is configured.
+              
+
+### 1.1.2 Backend mechanisms to support timezone/clock configuration and query 
+
+#### 1.1.2.1 add/delete timezone 
+This creates or deletes the CLOCK entry in the Redis ConfigDB.
+
+```
+   "CLOCK": {
+        "system": {
+            "timezone_name": "US/Hawaii"
+        }
+    }
+```
+
+A change in the CLOCK entry triggers hostcfgd to start the CLOCK configuration script which ensures the timezone and timestamp to be consistent across host and various containers.   
+         
+For the host:   
+         
+1. change the local timezone on the host via command "timedatectl set-timezone <new_timezone>"  
+   "local time" is now modified by linking the file /etc/localtime to the /usr/share/zoneinfo/<timezone_name> directory.      
+2. retart rsyslog to apply the updated time via command "systemctl restart rsyslog.service"    
+        
+For the containers:     
+         
+1. with bind mounts, container /etc/localtime is updated when host /etc/localtime is updated. /etc/localtime points to the current timezone under /usr/share/zoneinfo/<timezone_name>.   

```

Comment:
Hi Bing,

Can you please explain in detail about these bind mount. how to test that ?

Reply:
Here is a link about docker  bind mount. https://docs.docker.com/storage/bind-mounts/.

To test, you can go to the container (e.g docker exec -it mgmt-framework bash), and check the output of
ls -l /etc/localtime
ls -l /etc/timezone

You should see it is linked to the correct timezone.